app/controllers/spree/admin/images_controller_decorator.rb

```diff
@@ -30,5 +30,8 @@ def viewable_ids
       unless params[:image][:viewable_ids].present?
         params[:image][:viewable_ids] = [@product.master.id]
       end
+
+      # return value to ensure assignment inside set_variants
+      params[:image][:viewable_ids]
```

Comment:
Is this method being used for its return value, its side effects on `params`, or both? It's doing a lot of stuff and it's not clear exactly what it's _supposed_ to be doing, which is a bit exacerbated by making it private and not unit-testing.

I know we're mostly constrained here to the controller/action shape dictated by upstream, but do we have any wiggle room to clean things up here? My eyes were glazing over a bit by the fourth or fifth time I read `params[:image][:viewable_ids]`. :-)

Reply:
That's a good observation. I changed the method to only return the value and not have side-effects.